src/btree/bt_cursor.c

```diff
@@ -1478,14 +1478,20 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
             ret = __cursor_col_modify(cbt, NULL, WT_UPDATE_TOMBSTONE);
         } else if (__cursor_fix_implicit(btree, cbt)) {
             /*
-             * Creating a record past the end of the tree in a fixed-length column-store implicitly
-             * fills the gap with empty records, delete the record.
+             * We are deleting an implicitly created record (which exists because creating a record
+             * past the end of the tree in a fixed-length column-store implicitly fills the gap with
+             * empty records). But because this record is empty, deleting it is a no-op. But keep
+             * the return value as 0, because the record exists implicitly.
+             *
+             * We don't want to add a tombstone to an empty record, because doing so could create a
+             * problem during reconciliation: If this is a part of a prepared transaction, we would
+             * need to save the update so that we can then put it into the history store, but in
+             * this case, there is nothing to save.
              *
              * Correct the btree cursor's location: the search will have pointed us at the
              * previous/next item, and that's not correct.
              */
             cbt->recno = cursor->recno;
-            ret = __cursor_col_modify(cbt, NULL, WT_UPDATE_TOMBSTONE);
```

Comment:
Changing this feels like it's moving away from a core design decision in FLCS, I'm not super familiar with why it was done this way. We may want to ask around for an FLCS expert to review. I am curious, why fix it here and not in reconciliation?

Reply:
I don't think this breaks anything in FLCS, as the implicit records are already created at this point, but let's still ask.

I spent some time trying to figure out how to fix it in reconciliation, and couldn't figure it out. If I took care of one place where it broke reconciliation, it caused problems in another place, which tells me it would have to be a bigger fix.

Reply:
Yes, that's also how I understand this, but let me rephrase it my words to see if we are on the same page: As the record is implicitly created and has nothing in it, delete by itself has no effect—reading the value of the record before delete returns 0, and reading the value after delete also returns 0. And the record still exists after the removal, because it has been implicitly created. And calling remove on an implicitly created key returns 0 (as opposed to WT_NOTFOUND), because the record implicitly exists.